internal/sdk/crypto/crypto.go

```diff
@@ -0,0 +1,222 @@
+package crypto
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"crypto/rand"
+	"crypto/rsa"
```

Comment:
Not that it's has a considerable impact for the next 10 years (if that), but any reason not to use an elliptic curve?

Reply:
I don't think is a problem but I guess we should be willing to migrate control plane logic as well
cc @shyiko